evap/student/templates/student_index.html

```diff
@@ -81,18 +81,44 @@
+                                            {% else %}
+                                                <div data-toggle="tooltip" data-placement="left" class="disabled-tooltip" title="{% trans "We did not receive enough votes to publish the results of this course" %}"><a class="btn btn-sm btn-default" disabled="disabled">{% trans "Results" %}</a></div>
+                                            {% endif %}
+                                        {% endif %}
+                                    {% else %}
+                                        {% if course.preliminary_grade_documents.count > 0 and can_download_grades %}
+                                            <div class="btn-group">
+                                                <button type="button" class="btn btn-sm btn-default dropdown-toggle" data-toggle="dropdown" aria-expanded="false" {{ disable_if_archived }}>{% trans "Download grades" %} <span class="caret"></span></button>
+                                                <ul class="dropdown-menu" role="menu">
+                                                    {% for grade_document in course.preliminary_grade_documents %}
+                                                        <li><a href="{% url "grades:download_grades" grade_document.id %}">{{ grade_document.description }}</a></li>
+                                                    {% endfor %}
+                                                </ul>
+                                            </div>
                                         {% endif %}
-                                    {% elif course.state == 'inEvaluation' %}
-                                        {% if course in voted_courses %}
-                                            <div data-toggle="tooltip" data-placement="left" class="disabled-tooltip" title="{% trans "You already evaluated this course" %}"><a class="btn btn-sm btn-default" disabled="disabled">{% trans "Evaluate" %}</a></div>
-                                        {% else %}
-                                            <a href="{% url "student:vote" course.id %}" class="btn btn-sm btn-primary">{% trans "Evaluate" %}</a>
+                                        {% if course.state == 'inEvaluation' %}
+                                            {% if course in voted_courses %}
+                                                <div data-toggle="tooltip" data-placement="left" class="disabled-tooltip" title="{% trans "You already evaluated this course" %}"><a class="btn btn-sm btn-default" disabled="disabled">{% trans "Evaluate" %}</a></div>
+                                            {% else %}
+                                                <a href="{% url "student:vote" course.id %}" class="btn btn-sm btn-primary">{% trans "Evaluate" %}</a>
+                                            {% endif %}
                                         {% endif %}
```

Comment:
here the logic for the different buttons is quite mixed together although they could be separated.

how about 
```
{% if course.state == 'inEvaluation' %}
    vote button stuff

{% if course.state == 'evaluated' or course.state == 'reviewed' or course.state == 'published' %}
    some grade button stuff
{% else %}
    some other grade button stuff

{% if course.state == 'published' %}
    results button stuff
```